packages/core/src/wallet/IndyWallet.ts

```diff
@@ -467,17 +470,21 @@ export class IndyWallet implements Wallet {
    */
   public async createKey({ seed, keyType }: CreateKeyOptions): Promise<Key> {
     try {
+      // Ed25519 is supported natively in Indy wallet
       if (keyType === KeyType.Ed25519) {
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         //@ts-ignore
         const verkey = await this.indy.createKey(this.handle, { seed, crypto_type: 'ed25519' })
         return Key.fromPublicKeyBase58(verkey, keyType)
       }
 
-      if (keyType === KeyType.Bls12381g1 || keyType === KeyType.Bls12381g2) {
-        const blsKeyPair = await BbsService.createKey({ keyType, seed })
-        await this.storeKeyPair(blsKeyPair)
-        return Key.fromPublicKeyBase58(blsKeyPair.publicKeyBase58, keyType)
+      // Check if there is a signing key provider for the specified key type.
+      if (this.signingKeyProviderRegistry.hasProviderForKeyType(keyType)) {
```

Comment:
Isn't this check redundant as we do this in a try catch and use the `this.signingKeyProviderRegistry.getProviderForKeyType` method, which can error and checks whether it exists or not.

Reply:
Yeah sort of. I thought this way we can keep the custom error message that this key type isn't supported in the indy wallet (As a wallet can support key types itself, as well as using the singing key provider). Do you think it's better to remove it?

Reply:
No, I am okay with this. Just more curious behind the reasoning.